lib/ui/proposals/list/interactor/proposals_bloc.dart

```diff
@@ -1,68 +1,94 @@
 import 'package:hypha_wallet/ui/architecture/result/result.dart';
 import 'package:hypha_wallet/ui/profile/interactor/profile_data.dart';
 import 'package:hypha_wallet/ui/profile/usecases/fetch_profile_use_case.dart';
+import 'package:hypha_wallet/ui/proposals/filter/interactor/filter_proposals_bloc.dart';
 import 'package:hypha_wallet/ui/proposals/filter/interactor/filter_status.dart';
 import 'package:hypha_wallet/ui/proposals/list/interactor/get_proposals_use_case_input.dart';
 import 'package:hypha_wallet/ui/proposals/list/usecases/get_proposals_use_case.dart';
 
 part 'proposals_bloc.freezed.dart';
-
 part 'proposals_event.dart';
-
 part 'proposals_state.dart';
 
 class ProposalsBloc extends Bloc<ProposalsEvent, ProposalsState> {
   final GetProposalsUseCase _getProposalsUseCase;
   final FetchProfileUseCase _fetchProfileUseCase;
   final ErrorHandlerManager _errorHandlerManager;
-  List<DaoData> daos = [];
-  ProposalsBloc(this._getProposalsUseCase, this._fetchProfileUseCase, this._errorHandlerManager)
-      : super(const ProposalsState()) {
+
+  ProposalsBloc(this._getProposalsUseCase, this._fetchProfileUseCase, this._errorHandlerManager) : super(const ProposalsState()) {
     on<_Initial>(_initial);
+    on<_Load>(_loadProposalsForEvent);
   }
 
+  List<DaoData> daos = [];
   FilterStatus filterStatus = FilterStatus.active;
+  /// This value can be modified based on your preferences
+  final int first = 10;
+  late int offset;
 
   Future<void> _initial(_Initial event, Emitter<ProposalsState> emit) async {
     if (!event.refresh) {
       emit(state.copyWith(pageState: PageState.loading));
       filterStatus = event.filterStatus;
     }
 
-    await _fetchAndEmitProposals(emit, filterStatus);
+    await _fetchDaos(event, emit);
   }
 
-  Future<void> _fetchAndEmitProposals(Emitter<ProposalsState> emit, FilterStatus filterStatus) async {
-    // Fetch DAOs
+  Future<void> _fetchDaos(_Initial event, Emitter<ProposalsState> emit) async {
     final Result<ProfileData, HyphaError> profileResult = await _fetchProfileUseCase.run();
 
     if (profileResult.isValue && profileResult.asValue!.value.daos.isNotEmpty) {
       daos = profileResult.asValue!.value.daos;
-      // Fetch Proposals using the fetched DAOs
-      final Result<List<ProposalModel>, HyphaError> proposalsResult =
-          await _getProposalsUseCase.run(GetProposalsUseCaseInput(daos, filterStatus));
-
-      if (proposalsResult.isValue) {
-        // Emit both daos and proposals in one state
-        emit(state.copyWith(
-          pageState: PageState.success,
-          proposals: proposalsResult.asValue!.value,
-        ));
+
+      await _loadProposalsForEvent(event, emit);
+    } else {
+      final HyphaError error = profileResult.isError ? profileResult.asError!.error : HyphaError.api('Failed to retrieve DAOs');
+      await _errorHandlerManager.handlerError(error);
+      emit(state.copyWith(pageState: PageState.failure));
+    }
+  }
+
```

Comment:
do we really need to use "dynamic" for the event here? 

If we can know the class, please use the specific event class.

Reply:
Yes, we need to use `dynamic` because this function can be triggered through either the initial or load event.

Reply:
???

  Future<void> _fetchAndEmitProposals(ProposalsEvent event, Emitter<ProposalsState> emit, List<DaoData> selectedDaos, int retryCounter) async {


this is what I meant.

Reply:
Please change both instances

Reply:
Ah yes, you're right.